packages/amplify-util-mock/src/mockAll.ts

```diff
@@ -31,13 +36,25 @@ export async function mockAllCategories(context: any) {
       }
     }
     // Run the mock servers
+    const selectedMockableResources = await prompter.pick<'many', string>(
+      'Select the category',
+      mockableResources.map((r) => r.service).filter((value, index, self) => self.indexOf(value) === index),
+      {
+        returnSize: 'many',
+      },
+    );
```

Comment:
how does DX look like for this ?

Reply:
thank you for consistency and DX purposes changes the prompt display values from ['AppSync', 'S3', 'Lambda'] to ['GraphQL API','Storage','Function']
![image](https://user-images.githubusercontent.com/87995712/230805155-07caff2b-d185-4585-b335-c86c39d847f4.png)

Reply:
As far as I can read the code in `mockAll.ts` code `mockableResources` can contain multiple apis and functions.
What happens to this DX in that case? Should we list flat list of resources from each category or have 2 stage selection, i.e. first category and then expand all resources.
This DX becomes tricky if number of resources in each category is large, i.e. more than 10.

Reply:
we cant mock multiple api as a project can only have one graphql api, we dont mock rest api's.
we only mock graphql api, storage which is s3. additionally, i added functions as a option which was present in the file but never invoked(incomplete feature)

Reply:
That explains why assumptions and... shortcuts were made in current code...
What happens at this prompt if project has multiple functions? And what is desired DX?

Reply:
the DX would be as follows 
1. prompt user on the mockable categories present in the project 
2. user selects the categories
3. if function has been selected provide the function name for user to selec
4. start functions

https://user-images.githubusercontent.com/87995712/230987317-dd43e8a8-77a1-46fb-9859-7be2a1797a53.mov


5. start remaining categories